File: windows/QuotaArc/Model/ElapsedCopy.cs
namespace QuotaArc.Model;

internal static class ElapsedCopy
{
    public static string Ago(DateTime since, DateTime? now = null)
    {
        var elapsed = Text(since, now);
        return elapsed == "just now" ? elapsed : $"{elapsed} ago";
    }

    public static string Text(DateTime since, DateTime? now = null)
    {
        var n = now ?? DateTime.Now;
        var seconds = Math.Max(0, (n - since).TotalSeconds);
        if (seconds < 45) return "just now";

        var minutes = (int)Math.Round(seconds / 60);
        if (minutes < 60) return $"{Math.Max(1, minutes)} min";

        var hours = minutes / 60;
        var rest = minutes % 60;
        return rest == 0 ? $"{hours} hr" : $"{hours} hr {rest} min";
    }
}

File: Sources/Model/ElapsedCopy.swift
import Foundation

/// "how long has it been like this" — the second half of answering "is Claude
/// still working".
enum ElapsedCopy {
    /// The same span, phrased as a point in the past.
    static func ago(since: Date, now: Date = Date()) -> String {
        let elapsed = text(since: since, now: now)
        return elapsed == "just now" ? elapsed : "\(elapsed) ago"
    }

    static func text(since: Date, now: Date = Date()) -> String {
        let seconds = max(0, now.timeIntervalSince(since))
        if seconds < 45 { return "just now" }

        let minutes = Int((seconds / 60).rounded())
        if minutes < 60 { return "\(max(1, minutes)) min" }

        let hours = minutes / 60
        let rest = minutes % 60
        if rest == 0 { return "\(hours) hr" }
        return "\(hours) hr \(rest) min"
    }
}
